packages/utils-tests-helpers/src/hint-runner.ts

```diff
@@ -75,13 +75,25 @@ const validateResults = (t: GenericTestContext<Context<any>>, results: Problem[]
         return t.fail(`Result count is ${results.length}, should be ${reports.length}`);
     }
 
-    return reports.forEach((report, index: number) => {
-        if (server.port) {
-            t.is(results[index].message, report.message.replace(localhostRegex, `$1://localhost:${server.port}/`), `Different message`);
-        } else {
-            t.is(results[index].message, report.message, `Different message`);
-        }
-        if (report.position) {
+    if (server.port) {
+        reports.forEach((report) => {
+            report.message = report.message.replace(localhostRegex, `$1://localhost:${server.port}/`);
+        });
+    }
+
+    results.forEach((result) => {
+        const { message } = result;
+        let index = 0;
+        const found = reports.some((report, i) => {
+            index = i;
+            return report.message === message;
+        });
+
+        t.true(found, `No reports match "${message}"`);
+
+        const report = reports[index];
```

Comment:
We might want to check too that other properties are the same before removing (if they exist) just in case we are supposed to receive the same message multiple times but for different locations.